xlators/features/changelog/lib/src/gf-history-changelog.c

```diff
@@ -892,6 +892,7 @@ gf_history_changelog(char *changelog_dir, unsigned long start,
                        "unable to read htime file");
                 goto out;
             }
+            buffer[n_read] = '\0';
```

Comment:
If we have read exactly `MAX_PATH` bytes, this write will happen outside the buffer.

I think we should read `PATH_MAX - 1` bytes at line 887.

Reply:
Ack